### ISSUE TITLE
[Feature request] Add support for easy language switching

### ISSUE THREAD

**Issue by SoulInfernoDE** [mmtrt/cnctsun#3]
Hi,
i have made a language pack for german. It would be very nice if there is any option to include/download the language pack and overwrite existing files with it from within either the launcher or from the snap itself.
If you have any idea how this could be done the smart way i could give you the link to the language pack and the references.

Way to manually download the language pack via terminal:
`wget --no-check-certificate 'https://raw.githubusercontent.com/mzramna/easy-google-drive-downloader/master/download_google_drive_file.sh' -O download_google_drive_file.sh && chmod +x download_google_drive_file.sh && ./download_google_drive_file.sh '10gE3vUSUcjFWC0gxsnUbbLlZODffdydE'`

Also i don't know if it's allowed to add the uncut patches from Nyerguds but this would fit the game in german language nearly the same as in english and has a very nice play feeling.

**Comment by mmtrt** [comment]
I could implement this into the snap with following possible ways. I like the third option.

* detect user lang using on the distro if it matches GER then do the magic.
* add a var flag to snap when user run snap with flag then do the magic.
* add options on game type selection screen showing install game with unofficial GER translation.

can you post contents of tar file I have slow connection could take time, so the lang pack is for full game not for multiplayer of cncnet client if I'm correct.

**Comment by SoulInfernoDE** [comment]
Yes i think the third option would be a good solution. You are right it's a full game translation including the firestorm expansion. This includes a sound, text and video translation. You mentioned a good point that not all people have a fast internet connection. So adding a second option that only translates core parts of the game without the bigger files which excludes videos for example would be a good thing. I'll make a second archive excluding the bigger files.

First here are the folder containing all of the files including a screenshot of the contents:
[https://drive.google.com/drive/folders/1ExLoLW1hHbHxexHTJg4n8frwPPpUC7n7?usp=sharing](url)

[Full german language pack](https://drive.google.com/drive/folders/1iOg6ry28VIJwAhh9hvcNgvwc3oyNYA1m?usp=sharing) contents:
![content_of_cnctsf_uncut_language_pack_german_tar_bz2](https://user-images.githubusercontent.com/3682343/52161653-ce6bd980-26c8-11e9-9f93-a02ba9a05271.png)
Explanation of some files:
ecache81.mix - The Patch from Zimbio81 uncuts the cyborgs.
MOVIES01-03.MIX - translates the movies to german
TIBSUN.MIX ->All other files - translates everything possible with a language pack

[Small german language](https://drive.google.com/file/d/1yNuqd7v_mGIXaEWVipG_JKaCLO_JfRK_/view?usp=sharing) pack  - excluding any media contents:
![contents_of_cnctsf_uncut_language_pack_german_small_tar_bz2](https://user-images.githubusercontent.com/3682343/52161618-10485000-26c8-11e9-9c8e-25de17309799.png)

Manual download of the small language pack in terminal:
`wget --no-check-certificate 'https://raw.githubusercontent.com/mzramna/easy-google-drive-downloader/master/download_google_drive_file.sh' -O download_google_drive_file.sh && chmod +x download_google_drive_file.sh && ./download_google_drive_file.sh '1yNuqd7v_mGIXaEWVipG_JKaCLO_JfRK_' && rm download_google_drive_file.sh`

**Comment by mmtrt** [comment]
@SoulInfernoDE can you test the changes I've just added are living on edge channel.

`snap install --edge cnctsun`

or update the snap

`snap refresh --edge cnctsun`

**Comment by mmtrt** [comment]
Retest the snap from edge and confirm it.

**Comment by SoulInfernoDE** [comment]
commented [d0cfecc](https://github.com/mmtrt/cnctsun/commit/d0cfeccc98836a3fb50e898aa737b7d4c719f819)

**Comment by mmtrt** [comment]
That permission issue is harmless as for lang pack you shoud repack and upload it.

**Comment by SoulInfernoDE** [comment]
Hey mmtrt,
i tried to delete everything left on my computer and then reinstalled after i reuploaded the repackaged language pack. It works like a charm! THANK YOU! :1st_place_medal: 

_There are one or two notes we should give users who want to install the full game version:_
- They should  **NOT** select: "Install movies and audio" from the Installer as the files will be double downloaded first in english then in german and then it will be overwritten. So downloading the english versions isn't necessary anymore.
- The Installation of the language pack can take approx. ~6 minutes. This is due to the high compression of the language pack in .bz2-format. Packing in this format has good compatibility and saved approx. ~330 MB of downloading without compression.


So if you are willing to further support this i could upload a language pack for Spanish and French, too. So everyone will be pleased :-)

**Comment by mmtrt** [comment]
yeah do that then post their link ids, I could remove the silent install flag from lang pack install option so users could freely select files.

**Comment by SoulInfernoDE** [comment]
**_So the snap installation process of the full language packs will be:_**
- install the snap ->
- wine will be auto-configured ->
- install Tiberian Sun with TSInstaller but unselect "Install Movies" and unselect "Install Audio" in TSInstaller ->
- full language pack will download, extract and overwrites existing files.


**Language pack for french is ready:**
- [Core language pack](https://drive.google.com/file/d/1TqadPo4VYT3gsWAKOVGS3apALQVzXONO/view?usp=sharing) (without movies): - 1TqadPo4VYT3gsWAKOVGS3apALQVzXONO
- [Full language pack](https://drive.google.com/file/d/1Yo_tro6140ugdwrTLFF8tHnfoXuxEh7u/view?usp=sharing) (with movies) - 1Yo_tro6140ugdwrTLFF8tHnfoXuxEh7u

**Language pack for spanish is ready:**
- [Core language pack](https://drive.google.com/file/d/1JXcUHq0mgaOFhYMetYlhaTgtMH_7bYpI/view?usp=sharing) (without movies) - 1JXcUHq0mgaOFhYMetYlhaTgtMH_7bYpI
- [Full language pack](https://drive.google.com/file/d/1e8OugEos3sNQzobR01Dyn80OxZ9FB7cI/view?usp=sharing) (with movies) - 1e8OugEos3sNQzobR01Dyn80OxZ9FB7cI

_-> I recommend renaming the first entry: Full Game to something like: **[English] Full Game (Singleplayer + Multiplayer)** so users would recognize that there are options for their language available_
[      "Full Game (Singleplayer + Multiplayer)" "Multiplayer" "Core translation GER unofficial with full game" "Complete translation GER unofficial with full game")](https://github.com/mmtrt/cnctsun/blob/d0cfeccc98836a3fb50e898aa737b7d4c719f819/scripts/sommelier#L149)

**Comment by mmtrt** [comment]
can you test new changes though I'll add the notice warn users before install later after some testings.

**Comment by SoulInfernoDE** [comment]
Tested and commented commit [6e24c2b](https://github.com/mmtrt/cnctsun/commit/6e24c2b17faaac50f44e0ec865ad3e3ca5f88dad#commitcomment-32262455)
-> Only first option: "English - Full Game" seems to work yet.

**Comment by mmtrt** [comment]
https://github.com/mmtrt/cnctsun/commit/9b2d439bd0813d568126a44fb1b442f9dc3f734d should fix that issue.

**Comment by SoulInfernoDE** [comment]
I am testing right now all options from start to end. Adding final statement soon :+1:
